akka-actor-tests/src/test/scala/akka/util/TokenBucketSpec.scala

```diff
@@ -0,0 +1,253 @@
+/**
+ * Copyright (C) 2015-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.util
+
+import akka.testkit.AkkaSpec
+
+import scala.util.Random
+
+class TokenBucketSpec extends AkkaSpec {
+
+  class TestBucket(_cap: Long, _period: Long) extends TokenBucket(_cap, _period) {
+    var currentTime: Long = 0L
+  }
+
+  "A Token Bucket" must {
```

Comment:
Maybe it is time to write a tool for such cases ourselves? I am yet to figure out how can one be nicely readable and not more boilerplate than rolling your own (most of these tests rely on specific environmental help like dummy clocks, stream probes, etc) but this is something I would be happy to investigate.